# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/SurfaceTypes.h
# Commit: 204b899f436d
# Full Hash: 204b899f436d99f0eae759af3ccfcbaabfbbb428
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 09:34:54
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 3f5baab34b1f -r 204b899f436d gfx/gl/SurfaceTypes.h
--- a/gfx/gl/SurfaceTypes.h	Thu Jun 11 06:36:55 2020 +0000
+++ b/gfx/gl/SurfaceTypes.h	Thu Jun 11 06:37:35 2020 +0000
@@ -6,9 +6,7 @@
 #ifndef SURFACE_TYPES_H_
 #define SURFACE_TYPES_H_
 
-#include "mozilla/RefPtr.h"
-#include "mozilla/Attributes.h"
-#include <stdint.h>
+#include <cstdint>
 
 namespace mozilla {
 namespace layers {
@@ -17,58 +15,7 @@
 
 namespace gl {
 
-struct SurfaceCaps final {
-  bool any = false;
-  bool color = false;
-  bool alpha = false;
-  bool bpp16 = false;
-  bool depth = false;
-  bool stencil = false;
-  bool premultAlpha = true;
-  bool preserve = false;
-
-  // The surface allocator that we want to create this
-  // for.  May be null.
-  RefPtr<layers::LayersIPCChannel> surfaceAllocator;
-
-  SurfaceCaps();
-  SurfaceCaps(const SurfaceCaps& other);
-  ~SurfaceCaps();
-
-  void Clear() { *this = {}; }
-
-  SurfaceCaps& operator=(const SurfaceCaps& other);
-
-  // We can't use just 'RGB' here, since it's an ancient Windows macro.
-  static SurfaceCaps ForRGB() {
-    SurfaceCaps caps;
-
-    caps.color = true;
-
-    return caps;
-  }
-
-  static SurfaceCaps ForRGBA() {
-    SurfaceCaps caps;
-
-    caps.color = true;
-    caps.alpha = true;
-
-    return caps;
-  }
-
-  static SurfaceCaps Any() {
-    SurfaceCaps caps;
-
-    caps.any = true;
-
-    return caps;
-  }
-};
-
 enum class SharedSurfaceType : uint8_t {
-  Unknown = 0,
-
   Basic,
   EGLImageShare,
   EGLSurfaceANGLE,
@@ -79,21 +26,9 @@
   SharedGLTexture,
   AndroidSurfaceTexture,
   EGLSurfaceDMABUF,
-
-  Max
-};
-
-enum class AttachmentType : uint8_t {
-  Screen = 0,
-
-  GLTexture,
-  GLRenderbuffer,
-
-  Max
 };
 
 }  // namespace gl
+}  // namespace mozilla
 
-} /* namespace mozilla */
-
-#endif /* SURFACE_TYPES_H_ */
+#endif  // SURFACE_TYPES_H_